include/swift/IDE/Formatting.h

```diff
@@ -0,0 +1,107 @@
+//===--- Formatting.h -----------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_FORMATCONTEXT_H
+#define SWIFT_FORMATCONTEXT_H
+
+#include "swift/AST/AST.h"
+#include "swift/AST/ASTWalker.h"
+#include "swift/Frontend/Frontend.h"
+#include "swift/Basic/SourceManager.h"
+#include "swift/IDE/SourceEntityWalker.h"
+#include "swift/Subsystems.h"
```

Comment:
It's good practice to minimize includes in headers, possibly using forward declarations where appropriate.  I think most of these could be sunk into the .cpp file.